pkg/prometheus/operator.go

```diff
@@ -1630,15 +1630,18 @@ func (c *Operator) selectServiceMonitors(ctx context.Context, p *monitoringv1.Pr
 			smKey := fmt.Sprintf("serviceMonitor/%s/%s/%d", sm.GetNamespace(), sm.GetName(), i)
 
 			if err = store.addBearerToken(ctx, sm.GetNamespace(), endpoint.BearerTokenSecret, smKey); err != nil {
+				level.Error(c.logger).Log("error", err, "namespace", sm.GetNamespace(), "servicemonitor", sm.GetName())
```

Comment:
I wouldn't log anything here since we're already logging a warning after we break from the for loop.